File: app/src/main/java/com/sena/qfinder/QfinderApplication.java
package com.sena.qfinder;

import android.app.Application;
import android.util.Log;

import com.sena.qfinder.utils.AppLifecycleHandler;
import com.sena.qfinder.utils.FirebaseInitializer;

public class QfinderApplication extends Application {
    private static final String TAG = "QFinderApplication";

    @Override
    public void onCreate() {
        super.onCreate();

        // Inicializar Firebase
        FirebaseInitializer.initialize(this);

        // Registrar el lifecycle handler
        registerActivityLifecycleCallbacks(new AppLifecycleHandler());

        Log.d(TAG, "Application initialized");
    }
}

File: app/QfinderApplication.kt
import android.app.Application
import android.util.Log
import com.google.firebase.FirebaseApp

class QfinderApplication : Application() {
    override fun onCreate() {
        super.onCreate()
        Log.d("QfinderApplication", "Initializing Firebase")
        try {
            FirebaseApp.initializeApp(this)
            Log.d("QfinderApplication", "Firebase initialized successfully")
        } catch (e: Exception) {
            Log.e("QfinderApplication", "Failed to initialize Firebase", e)
        }
    }
}